src/api/test-controller/index.js

```diff
@@ -376,9 +376,22 @@ export default class TestController {
         return this.testRun.executeAction(name, new GetBrowserConsoleMessagesCommand(), callsite);
     }
 
+    _checkForExessiveAwaits (selectorCallsiteList, expectCallsite) {
+        selectorCallsiteList.forEach(selectorCallsite => {
+            if (selectorCallsite.filename === expectCallsite.filename &&
+                selectorCallsite.lineNum === expectCallsite.lineNum) {
+                this._addWarning(WARNING_MESSAGE.exessiveAwaitInAssertion, selectorCallsite);
+
+                selectorCallsiteList.delete(selectorCallsite);
+            }
+        });
```

Comment:
I feel that the modern `for..of` loop is better for almost any situation when we had to use the good old `forEach` before. I think if you don't write your code in the pure functional style or don't use long functional chains (`.map().reduce().filter().forEach()`), `for..of` is preferred.

```suggestion
        for (const selectorCallsite of selectorCallsiteList) {
            if (selectorCallsite.filename === expectCallsite.filename &&
                selectorCallsite.lineNum === expectCallsite.lineNum) {
                this._addWarning(WARNING_MESSAGE.exessiveAwaitInAssertion, selectorCallsite);

                selectorCallsiteList.delete(selectorCallsite);
            }
        }
```